firebase-firestore/src/test/java/com/google/firebase/firestore/remote/RemoteSerializerTest.java

```diff
@@ -680,6 +682,85 @@ public void testEncodesMultipleFiltersOnDeeperCollections() {
+                            .addFilters(
+                                Filter.newBuilder()
+                                    .setCompositeFilter(
+                                        StructuredQuery.CompositeFilter.newBuilder()
+                                            .setOp(CompositeFilter.Operator.AND)
+                                            .addFilters(
+                                                Filter.newBuilder()
+                                                    .setFieldFilter(
+                                                        StructuredQuery.FieldFilter.newBuilder()
+                                                            .setField(
+                                                                FieldReference.newBuilder()
+                                                                    .setFieldPath("author"))
+                                                            .setOp(Operator.EQUAL)
+                                                            .setValue(
+                                                                Value.newBuilder()
+                                                                    .setStringValue("dimond"))))
+                                            .addFilters(
+                                                Filter.newBuilder()
+                                                    .setFieldFilter(
+                                                        StructuredQuery.FieldFilter.newBuilder()
+                                                            .setField(
+                                                                FieldReference.newBuilder()
+                                                                    .setFieldPath("tags"))
+                                                            .setOp(Operator.ARRAY_CONTAINS)
+                                                            .setValue(
+                                                                Value.newBuilder()
+                                                                    .setStringValue(
+                                                                        "pending"))))))))
+            .addOrderBy(
+                Order.newBuilder()
+                    .setField(FieldReference.newBuilder().setFieldPath("prop"))
+                    .setDirection(Direction.ASCENDING))
+            .addOrderBy(defaultKeyOrder());
+    QueryTarget.Builder queryBuilder =
+        QueryTarget.newBuilder()
+            .setParent("projects/p/databases/d/documents/rooms/1/messages/10")
+            .setStructuredQuery(structuredQueryBuilder);
+    Target expected =
+        Target.newBuilder()
+            .setQuery(queryBuilder)
+            .setTargetId(1)
+            .setResumeToken(ByteString.EMPTY)
+            .build();
+
+    assertEquals(expected, actual);
+    assertEquals(
+        serializer.decodeQueryTarget(serializer.encodeQueryTarget(q.toTarget())), q.toTarget());
```

Comment:
Maybe follow the existing pattern and create an intermediary variable "roundTripped" so that we can debug serialization failures better.